### PR TITLE
Unified test input files

### PR DESCRIPTION
**Pull request by ijlee2**
## Description

This PR addresses issue #12.

## What changed?

1\. I moved the input and output files to follow the folder structure,

```
ember-component-template-colocation-migrator
│   ...
│
├── test
│   └── fixtures
│       ├── example-js
│       │   ├── input.js
│       │   ├── classic-to-flat.js
│       │   └── classic-to-nested.js
│       │ 
│       ├── example-ts
│       │   ├── input.js
│       │   ├── classic-to-flat.js
│       │   └── classic-to-nested.js
│   ...
```

2\. I separated the existing tests in `test/migrator-test.js` into 2 files:

- `test/lib/migrator/classic-to-flat-test.js`
- `test/lib/migrator/classic-to-nested-test.js`

This was done so that (1) we can easily set the locations of input and output files, and (2) the test descriptions aren't auto-generated and can be more meaningful:
```javascript
// Before
var fixturesPath = path.resolve(__dirname, "fixtures/classic-to-flat");
var entries = fse.readdirSync(fixturesPath);    

entries.forEach(async function(entry) { 
  it(`should migrate ${entry} fixture properly`, async function() { 
    var fixturePath = path.join(fixturesPath, entry);   
    var input = require(fixturePath + "/input");    
    var expected = require(fixturePath + "/output");


// After
describe('For an app with component classes written in JavaScript', function() {
  beforeEach(function() {
    const fixturePath = path.resolve(__dirname, '../../fixtures/example-js');
    const input = require(`${fixturePath}/input`);
    this.expectedOutput = require(`${fixturePath}/classic-to-flat`);
```

Having the nested folder `test/lib/migrator` may feel excessive (versus just `test/migrator`). I was hoping that by creating `test/lib`, someone may want to write tests for the files in `lib/utils`.

3\. I fixed a bug that was uncaught in PR https://github.com/ember-codemods/ember-component-template-colocation-migrator/pull/11. The new tests showed that partial templates are incorrectly moved to `app/components` when we run the codemod twice.

4\. I removed `power-assert`, which wasn't used by tests, and updated the `README` file.

### CODE REVIEW

File: lib/migrator.js [hunk 1]
```diff
@@ -27,8 +27,9 @@ module.exports = class Migrator {
   }
 
   findTemplates() {
-    const templateFolderPath = path.join(this.projectRoot, 'app/templates');
-    const templateFilePaths = glob.sync(`${templateFolderPath}/**/*.hbs`);
+    const componentsFolderPath = path.join(this.projectRoot, 'app/components');
```
Comment (ijlee2) [inline lib/migrator.js:30]: In a test, the first time that the codemod ran, the component template `partials/with-partial.hbs` moved from `app/templates` to `app/components`. So, the second time it runs, I also needed to look at `app/components` to see if there are any components that use a partial.

File: README.md [hunk 2]
```diff
@@ -53,6 +57,12 @@ your-project-name
 ```
 
 
-### Running Tests
+## Running Tests
+
+The tests were inspired by [ember-module-migrator](https://github.com/ember-codemods/ember-module-migrator/blob/master/test/engines/classic-test.js).
```
Comment (ijlee2) [inline README.md:62]: Moved the comment that was in `test/migrator-test.js` (now deleted) to here.

**Review by rwjblue — APPROVED** [review]
Awesome thank you @ijlee2!

**Review by rwjblue — APPROVED** [review]
Awesome thank you @ijlee2!

**Pull request merged**